modules/generator/processor/util/util.go

```diff
@@ -68,21 +70,37 @@ func GetJobValue(attributes []*v1_common.KeyValue) string {
 	return namespace + svName
 }
 
-func GetTargetInfoAttributesValues(attributes []*v1_common.KeyValue, exclude []string) ([]string, []string) {
+func GetTargetInfoAttributesValues(attributes []*v1_common.KeyValue, exclude, intrinsicLabels []string) ([]string, []string) {
 	// TODO allocate with known length, or take new params for existing buffers
-	keys := make([]string, 0)
-	values := make([]string, 0)
+	keys := make([]string, 0, len(attributes))
+	values := make([]string, 0, len(attributes))
+	count := 0
 	for _, attrs := range attributes {
 		// ignoring job and instance
 		key := attrs.Key
-		value := tempo_util.StringifyAnyValue(attrs.Value)
 		if key != "service.name" && key != "service.namespace" && key != "service.instance.id" && !Contains(key, exclude) {
-			keys = append(keys, key)
+			keys = append(keys, SanitizeLabelNameWithCollisions(key, intrinsicLabels))
+			value := tempo_util.StringifyAnyValue(attrs.Value)
 			values = append(values, value)
+			count++
 		}
 	}
 
-	return keys, values
+	return keys[:count], values[:count]
+}
+
+func SanitizeLabelNameWithCollisions(name string, dimensions []string) string {
+	sanitized := strutil.SanitizeLabelName(name)
+
+	if IsIntrinsicDimension(sanitized, dimensions) {
```

Comment:
Can probably replace all of IsIntrinsicDimension with [slices.Contains](https://pkg.go.dev/slices#Contains)